### ISSUE TITLE
`stable-diffusion-webui` plugin

### ISSUE THREAD

**Issue by jtydhr88** [Stability-AI/StableStudio#25]
https://github.com/Stability-AI/StableStudio/pull/24

**Comment by wendaleruan** [comment]
Usage Error: Couldn't find a script named "dev:use-webui-plugin".
how to fix it @jtydhr88

**Comment by cruhl** [comment]
@jtydhr88 In terms of what’s needed to merge, I think we’d be good to go once this plug-in is at feature parity with the Stability API plugin with the exception of storing images.

I think we can add storing images locally as a follow-up if you wanted 👍

**Comment by harrywang** [comment]
@jtydhr88 "MacOS and Linux support (since I don’t have environment with MacOS/Linux, may need someone help with this" I am a Mac user and can help with this. Please let me knwo what to do.

**Comment by jtydhr88** [comment]
Thanks @harrywang ! Just make some normal actions I guess you already made would be helpful, and create issues if find any bugs

**Comment by jtydhr88** [comment]
https://github.com/Stability-AI/StableStudio/pull/70

**Comment by KaiW98** [comment]
hello, I inserted stable-diffusion-webui according to the method provided in the linux environment. Adding the url in stablestudio keeps turning around, and the insertion does not make progress. How to solve this problem?
![image](https://github.com/Stability-AI/StableStudio/assets/74865581/b02f35af-71e7-475f-8c65-77c3e8bc6a9b)
![image](https://github.com/Stability-AI/StableStudio/assets/74865581/bfee02d1-2a9a-4328-98e0-20f95914ef4f)
![image](https://github.com/Stability-AI/StableStudio/assets/74865581/efa73420-dff9-4481-b3c7-c7eff5aa1902)

**Comment by jtydhr88** [comment]
@KaiW98 see here
![image](https://github.com/Stability-AI/StableStudio/assets/860985/110eeb9a-e292-410f-afe7-25e6ffa917d0)
on your webui side, you should use **http://localhost:3000** instead of http://127.0.0.1:3000 in cors-allow-orgins
btw, this link should map your StableStudio log:
![image](https://github.com/Stability-AI/StableStudio/assets/860985/1bef007c-3d15-43d0-81da-92092128cb78)

**Comment by harrywang** [comment]
> @KaiW98 see here ![image](https://user-images.githubusercontent.com/860985/241733881-110eeb9a-e292-410f-afe7-25e6ffa917d0.png) on your webui side, you should use **http://localhost:3000** instead of http://127.0.0.1:3000 in cors-allow-orgins btw, this link should map your StableStudio log: ![image](https://user-images.githubusercontent.com/860985/241734256-1bef007c-3d15-43d0-81da-92092128cb78.png)

@jtydhr88 https://github.com/Stability-AI/StableStudio/issues/71 I reported the same issue and figured it out myself - should we add some instruction in the README?

**Comment by jtydhr88** [comment]
@harrywang looks like @KaiW98 is reverse to yours, in your case, you configured localhost:3000 but didn't work, but on his case, I think he need to configure localhost:3000.

**Comment by nmcbride** [comment]
I'm on the latest versions of both as of today.

**Comment by jtydhr88** [comment]
hey @nmcbride can you share which version of webui, for mine is on 1.3.0 commit:
![image](https://github.com/Stability-AI/StableStudio/assets/860985/7f8408df-a59a-4678-8da5-7bf40b64301e)

**Comment by KaiW98** [comment]
I also found the same problem, the webui version is v1.3.0.
Modify the cors-allow-orgins address according to the above method, still not working
![image](https://github.com/Stability-AI/StableStudio/assets/74865581/8a5343bc-0d50-4341-adea-3b24c226994c)
![image](https://github.com/Stability-AI/StableStudio/assets/74865581/59a1b9d3-36bc-4740-9c87-f7a92dee9a75)
![image](https://github.com/Stability-AI/StableStudio/assets/74865581/2f53e9f6-1235-4249-918e-57d8e98630fc)

**Comment by GRMrGecko** [comment]
I received the same error until I followed the instructions and installed the plugin: https://github.com/jtydhr88/sd-webui-StableStudio

**Comment by jtydhr88** [comment]
hey guys, I made a quick bug fix https://github.com/Stability-AI/StableStudio/pull/76 here but I don't think it will fix your guys issue, still research how to reproduce the bug you are facing...

**Comment by nmcbride** [comment]
I made process but I think there may be an issue with my stable diffusion even though it is the latest version and  works from it's own web UI.

When img2text is called it throws a 404 with the following:
```{"error":"HTTPException","detail":"Sampler not found","body":"","errors":""}```

**Comment by GRMrGecko** [comment]
You need to select a sampler in the advanced options above the dream button.

**Comment by nmcbride** [comment]
I see. I was thrown off because I also tried the call from stable diffusion's api page using their example body and it threw the same error. So I was figuring the problem was bigger than I thought.

**Comment by jtydhr88** [comment]
The difficulty for me is I have no idea how to reproduce this issue, I am happy if someone could provide more information to help me to reproduce it

**Comment by s-blu** [comment]
Hello,

thanks alot for this plugin! Just FYI: I had the problem that StableStudio reported a CORS Error and stable-diffusion-webui ran into a 500 with a `1 validation error for Options\nresponse -> sd_model_checkpoint\n  value is not None (type=type_error.not_none)` stack trace when trying to access `/sdapi/v1/options`. I was able to resolve this issue by starting up the automatic1111 UI and disabling the installed ControlNet extension. It might be worthwhile to check for and disable extensions installed via automatic1111 if this plugin is not running as expected.

**Comment by jtydhr88** [comment]
@s-blu great catch! That information may help me to reproduce the issue!

**Comment by nmcbride** [comment]
For me the CORS issue was just not defining it as arguments. It was 100% on my end.
After defining CORS and installing the plugin and setting the values in the UI, everything worked.

**Comment by maekawataiki** [comment]
I also faced the error `1 validation error for Options\nresponse -> sd_model_checkpoint\n  value is not None (type=type_error.not_none)`

**TLDR**: creating `config.json` with following command solves the error
`echo "{\"sd_model_checkpoint\": \"\", \"sd_checkpoint_hash\": \"\"}" > stable-diffusion-webui/config.json`

Response model of `/options` endpoint is set to OptionsModel ([ref](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/baf6946e06249c5af9851c60171692c44ef633e0/modules/api/models.py#L218)) and type of each attribute of OptionsModel is set to type of value of opts on initialization ([ref](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/baf6946e06249c5af9851c60171692c44ef633e0/modules/api/models.py#L207-L218)) which can be overriden by file `stable-diffusion-webui/config.json` ([ref](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/baf6946e06249c5af9851c60171692c44ef633e0/modules/shared.py#L697-L699)).

The error is caused when app is initialized with `sd_model_checkpoint` set to default value of None ([ref](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/baf6946e06249c5af9851c60171692c44ef633e0/modules/shared.py#L405)) which result in FastAPI expecting `NoneType` for `sd_model_checkpoint`.

Disabling extension is not necessary but clicking "Apply and restart" is the important part since it writes current options to `config.json` (`sd_model_checkpoint` is set to i.e. `"v1-5-pruned-emaonly.safetensors [6ce0161689]"`). After that, newly launched application loads the file on initialization and FastAPI expects string for `sd_model_checkpoint` and error is solved. Running command `echo "{\"sd_model_checkpoint\": \"\", \"sd_checkpoint_hash\": \"\"}" > stable-diffusion-webui/config.json` to populate default value can also solve the error.
